src/transactions.js

```diff
@@ -0,0 +1,171 @@
+import * as util from './util';
+import {DERIVATION, LENS, REACTOR} from './types';
+import {UNKNOWN, UNCHANGED, CHANGED} from './states';
+
+export function mark (node, reactors) {
+  for (var i = 0, len = node._activeChildren.length; i < len; i++) {
+    var child = node._activeChildren[i];
+    switch (child._type) {
+      case DERIVATION:
+      case LENS:
+        if (child._state !== UNKNOWN) {
+          child._state = UNKNOWN;
+          mark(child, reactors);
+        }
+        break;
+      case REACTOR:
+        reactors.push(child);
+        break;
+    }
+  }
+}
+
+export function processReactors (reactors) {
+  for (var i = 0, len = reactors.length; i < len; i++) {
+    var r = reactors[i];
+    if (r._reacting) {
+      throw new Error("Synchronous cyclical reactions disallowed. " +
+                      "Use setImmediate.");
+    }
+    r._maybeReact();
+  }
+}
+
+var TransactionAbortion = {};
+
+function initiateAbortion() {
+  throw TransactionAbortion;
+}
+
+function TransactionContext(parent) {
+  this.parent = parent;
+  this.id2originalValue = {};
+  this.modifiedAtoms = [];
+}
+
+export function maybeTrack (atom) {
+  if (currentCtx !== null) {
```

Comment:
there is a function below called inTransaction() that you can use.
Hope u don't mind me commenting while i read through your code :)